src/Microsoft.Extensions.Caching.Memory/IMemoryCacheEvictionStrategy.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    public interface IMemoryCacheEvictionStrategy
+    {
+        // TODO: doc comments
+        IEnumerable<CacheEntry> GetEntriesToEvict(IEnumerable<CacheEntry> entries, DateTimeOffset now);
```

Comment:
Yea although this is the most generic collection, it may be a little too limited in terms of operations that are available on it.